### DISCUSSION TITLE
[Feature] DecSync support

### DISCUSSION THREAD

**Discussion by DominiqueSite** [TechbeeAT/jtxBoard#692] [Ideas]
**Is my feature request related to a problem?**
I use [DecSync](https://github.com/39aldo39/DecSync) for synchronizing tasks to my desktop PC with end-to-end encryption (via [Syncthing](https://syncthing.net/)). jtxBoard currently only supports options that require centralized servers for synchronization (which are also not end-to-end encrypted).

**The solution I'd like:**
I would like jtxBoard to integrate with [DecSync CC](https://github.com/39aldo39/DecSyncCC), the Android client for DecSync.

**Alternatives I've considered:**
jtxBoard could develop its own way of using a synchronizable local directory as its database.

**Comment by patrickunterwegs** [comment]
Hey @DominiqueSite 
I checked this DeckSync CC. Basically sich am implementation needs to be done in this app itself. jtx Board does offer a 'contract' which defines the fields and how to access them as well as ical4android interfaces to get an iCalendar File out of it. So only minor changes would be necessary in jtx Board, but a new implementation in DeckSync.

However, I'm not such a fan of the approach of this app. 
What I could rather imagine is a regular export/import job that puts the all collections as files in a directory regularly and and also imports them regularly. But it's rather complex with the conflict management... Like when there's a new entry in the database that is not in the file, does it mean it was deleted on another client or was it just not inserted yet? This whole sync management is fine by DAVx5 through the server sync. Such a logic would be necessary in jtx Board as well. That's pretty much effort and I'm not sure if it's really worth it. 
So I'm just saying to point out the complexity. For me this wouldn't be a top priority for now unless it gets a lot of attention by different users...

**Comment by patrickunterwegs** [comment]
I'm moving this topic to discussions as the implementation needs to be done in DeckSync CC.
